crates/sui-core/src/authority.rs

```diff
@@ -1033,15 +1046,11 @@ impl AuthorityState {
 
     pub async fn notify_read_effects(
         &self,
-        certificate: &VerifiedCertificate,
-    ) -> SuiResult<TransactionEffects> {
-        let tx_digest = *certificate.digest();
-        Ok(self
-            .database
-            .notify_read_executed_effects(vec![tx_digest])
-            .await?
-            .pop()
-            .expect("notify_read_effects should return exactly 1 element"))
+        digests: &[TransactionDigest],
```

Comment:
I don't quite understand the change to this function. It seems a lot of unnecessary churn. All the callers now must pass a newly created array on the fly, while previously it was much more convenient.

Reply:
The reason I did this was because of this https://github.com/MystenLabs/sui/pull/15891/files/26e4b6a044ff0ed1c2a08f445ef2a91ce5c36770#diff-7c1928de74ceda12b20305e1c01c40146d95ee3901f4beb00e6d2d5ed51e3fd4L305

So it was either change everything to use a multi-get interface, or add another method so that we had both single and multi-get options. And since the trait method in EffectsNotifyRead takes a slice, it seemed better to make everything consistent so that you never have to remember whether you're calling the single- or multi-get version.

But I don't feel strongly about it one way or the other, happy to change it back and add the second method if you still think its better after understanding the motivation

Reply:
So previously, we have a multi-read interface on AuthorityStore, and then an interface in AuthorityState taking &VerifiedCertificate.
Now we still have a multi-get interface on ExecutionCacheRead, but also a multi-get interface in AuthorityState.
Would it not work if the function in AuthorityState still takes &VerifiedCertificate?

Reply:
Reverted to the original code